### ISSUE TITLE
Request.sendAsync() keeps process running even after all other statements are completed

### ISSUE THREAD

**Issue by claudemiller** [neow3j/neow3j#54]
When calling `Request.sendAsync()` the process takes about a minute to terminate, even thought the response has already been received and processed.

Example code:
```
Neow3j neow3j = Neow3j.build(new HttpService("http://nucbox.axlabs.com:30333"));
NeoGetAccountState getAccountState = neow3j
                .getAccountState("AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y")
                .sendAsync().get();
```

**Comment by gsmachado** [comment]
I investigated this a bit, and I don't think it's a bug per se.

The thing is: The `run()` method in `Async` class completes normally and successfully. For async calls, we are using the executor (`ExecutorService`) with `newCachedThreadPool()` policy, and that is causing the main thread -- in this case, the developer's program -- to remain active until all threads are gone from the cache. As stated [here](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#newCachedThreadPool--), "threads that have not been used for sixty seconds are terminated and removed from the cache".

The `newCachedThreadPool()` executor is used because we assume that developers using neow3j might execute many short-lived asynchronous tasks, and, therefore, a cached thread pool helps to enhance performance.

My suggestion would be to provide a way to give an `Executor` instance to the `Neow3j` class to execute async calls. This way developers would have complete control over the executor's thread pool, if desired.

**Comment by gsmachado** [comment]
In the commits [68fcbe5](https://github.com/neow3j/neow3j/commit/68fcbe5e0b5aacb3f991d44a1f1feb61877acb99) and [2cc3bc9](https://github.com/neow3j/neow3j/commit/2cc3bc96db3c837691de9198af9d3382268252fb) I added a way to specify an external `ExecutorService`.
This gives a lot of flexibility for developers who want to control the lifecycle of the asynchronous threads, etc.

So, for documentation purposes, you could use it this way:

```java
    public static void main(String[] args) throws ExecutionException, InterruptedException {

        ExecutorService executorService = Executors.newFixedThreadPool(1);

        Neow3j neow3j = Neow3j.build(new HttpService("http://nucbox.axlabs.com:30333", executorService));
        NeoGetAccountState getAccountState = neow3j
                .getAccountState("AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y")
                .sendAsync().get();

        System.out.println(getAccountState.getAccountState());
        executorService.shutdown();
    }
```

Pay attention to the `executorService.shutdown()` in the end. This is required. Otherwise, the `main()` method will continue to wait until all threads are dead, indefinitely. This way, developers can, for example, attach a `shutdownHook()` to the main thread:

```java
    private static ExecutorService executorService = Executors.newFixedThreadPool(1);

    public static void main(String[] args) throws ExecutionException, InterruptedException {

        Neow3j neow3j = Neow3j.build(new HttpService("http://nucbox.axlabs.com:30333", executorService));
        NeoGetAccountState getAccountState = neow3j
                .getAccountState("AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y")
                .sendAsync().get();
        System.out.println(getAccountState.getAccountState());

        // Just terminate the program (sending a graceful signal to it)
        System.exit(0);
    }

    static {
        Runtime.getRuntime().addShutdownHook(new Thread() {
            public void run() {
                System.out.println("Shutting down...");
                executorService.shutdownNow();
            }
        });
    }
```

**Comment by gsmachado** [comment]
This is published on the `2.1.0-SNAPSHOT`. It will be included on the `2.1.0` release.
